jbpm-services/jbpm-services-ejb/jbpm-services-ejb-timer/src/main/java/org/jbpm/services/ejb/timer/EjbSchedulerService.java

```diff
@@ -129,43 +134,74 @@ public void afterCompletion(int status) {
         }
     }
 
-    private TimerJobInstance getTimerJobInstance (String uuid) {
+    private TimerJobInstance getTimerJobInstance(String uuid) {
         return unwrapTimerJobInstance(getEjbTimer(getTimerMappinInfo(uuid)));
     }
 
-
     @Override
     public TimerJobInstance getTimerJobInstance(long processInstanceId, long timerId) {
-        return unwrapTimerJobInstance(getEjbTimer(getTimerMappinInfo(processInstanceId, timerId)));
+        try {
+            return unwrapTimerJobInstance(getEjbTimer(getTimerMappinInfo(processInstanceId, timerId)));
+        } catch (Exception e) {
+            logger.warn("Problem retrieving timer for processInstance Id {}, returning null", timerId,
+                    processInstanceId,
+                    e);
+            return null;
+        }
     }
 
     private Timer getEjbTimer(TimerMappingInfo timerMappingInfo) {
-        try {
-            if(timerMappingInfo == null || timerMappingInfo.getInfo() == null) {
-                return null;
-            }
-            byte[] data = timerMappingInfo.getInfo();
-            return ((TimerHandle) new ObjectInputStream(new ByteArrayInputStream(data)).readObject()).getTimer();
-        } catch (Exception e) {
-            logger.warn("wast not able to deserialize info field from timer info for uuid");
+        if (timerMappingInfo == null || timerMappingInfo.getInfo() == null) {
             return null;
         }
+        try {
+            return ((TimerHandle) new ObjectInputStream(
+                    new ByteArrayInputStream(timerMappingInfo.getInfo()))
+                    .readObject()).getTimer();
+
+        } catch (ClassNotFoundException | IOException e) {
+            logger.warn("Error {} deserializing the timer handle for timermappinginfo {}", e.getMessage(),
+                    timerMappingInfo);
+            deleteTimerMapping(timerMappingInfo);
+            throw new RuntimeException(e);
```

Comment:
I don't think this method should throw an exception if something goes wrong during deserialization.

Reply:
yes, you are right, I changed it after a self-review already